include/xen/generic.h

```diff
@@ -11,4 +11,27 @@
 #define XEN_PAGE_SIZE		4096
 #define XEN_PAGE_SHIFT		12
 
+#define PFN_UP(x)		(unsigned long)(((x) + XEN_PAGE_SIZE-1) >> XEN_PAGE_SHIFT)
+#define PFN_DOWN(x)		(unsigned long)((x) >> XEN_PAGE_SHIFT)
+#define PFN_PHYS(x)		((unsigned long)(x) << XEN_PAGE_SHIFT)
+#define PHYS_PFN(x)		(unsigned long)((x) >> XEN_PAGE_SHIFT)
+
+#define to_phys(x)		((unsigned long)(x))
+#define to_virt(x)		((void *)(x))
+
+#define virt_to_pfn(_virt)	(PFN_DOWN(to_phys(_virt)))
```

Comment:
I don't quite understand the difference between virt_to_pfn() and virt_to_mfn(). Can you clarify it?

That said, it doesn't seem to be used. So this could possibly be dropped.

Reply:
That was taken from Mini-OS and after your clarification I see that this is redundant. Removed with single xen_virt_to_gfn().